interface/SimpleCacheSentry.h

```diff
@@ -27,14 +27,16 @@ class SimpleCacheSentry : public RooAbsArg {
         virtual RooAbsArg *createFundamental(const char* newname=0) const { return 0; }
         virtual Bool_t readFromStream(std::istream& is, Bool_t compact, Bool_t verbose=kFALSE) { return false; }
         virtual void writeToStream(std::ostream& os, Bool_t compact) const { }
+        virtual Bool_t operator==(const RooAbsArg& other) const { return this == &other; }
         virtual Bool_t operator==(const RooAbsArg& other) { return this == &other; }
         virtual void syncCache(const RooArgSet* nset=0) {}
         virtual void copyCache(const RooAbsArg* source, Bool_t valueOnly=kFALSE, Bool_t setValDirty=kTRUE) {}
         virtual void attachToTree(TTree& t, Int_t bufSize=32000) {}
         virtual void attachToVStore(RooVectorDataStore& vstore) {}
         virtual void setTreeBranchStatus(TTree& t, Bool_t active) {}
         virtual void fillTreeBranch(TTree& t) {}
-	virtual Bool_t isIdentical(const RooAbsArg& other, Bool_t assumeSameType=kFALSE) ;
+        virtual Bool_t isIdentical(const RooAbsArg& other, Bool_t assumeSameType=kFALSE) const;
+        virtual Bool_t isIdentical(const RooAbsArg& other, Bool_t assumeSameType=kFALSE) { return const_cast<SimpleCacheSentry*>(this)->isIdentical(other, assumeSameType); };
```

Comment:
Perhaps only one question here: Why is there a need for a non-const function to check if the caches are identical? Also the same comment for the operator ==.

Reply:
Const-qualification checking happens after overload resolution, so you have to include both const and non-const signatures if you want this to compile for both 6.22 and 6.18 (they added `const` to these virtual methods in the base class in 6.22).

Reply:
I understand, but the non-const version should actually never be used for simple comparison operators/functions (and I don't see where they are used in src/). const_cast is usually not good practice either, so if the code compiles without the non-const version of these overloads, I would suggest removing them in the first place.

Reply:
I agree, and you'll notice I added a const version. It won't compile without a non-const signature in 6.18 because the older ROOT base class has chosen a non-const a pure virtual signature.

Reply:
I could do as I did for the ROOT bug workaround elsewhere in this PR and use a preprocessor macro to switch them, but the net effect is the same.

Reply:
I was actually about to suggest that. I agree a preprocessor if statement could prevent the use of this non-const usage beyond 6.18.

Reply:
Given that once we are ready to make a release available for CMSSW_11_2 we will switch branches anyway that seems like overkill to me (but happy to take opinions from other combine devs)

Reply:
Yes, it's fine to leave it as implemented for now. Better not to have more preprocessor logic than needed.

Reply:
Since we're on 112x should I remove the 102x compat?